mlrun/runtimes/sparkjob/abstract.py

```diff
@@ -307,7 +307,13 @@ def _get_gpu_type_and_quantity(resources):
 
     def _validate(self, runobj: RunObject):
         # validating length limit for sparkjob's function name
-        verify_field_regex("run.metadata.name", runobj.metadata.name, sparkjob_name)
+        try:
+            verify_field_regex("run.metadata.name", runobj.metadata.name, sparkjob_name)
+
+        except mlrun.errors.MLRunInvalidArgumentError:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                f"Job name '{runobj.metadata.name}' is not valid. The job name must be not longer than 29 characters."
```

Comment:
There is no need for `.` in the end of an error message.
```suggestion
                f"Job name '{runobj.metadata.name}' is not valid. The job name must be not longer than 29 characters"
```